app/controllers/internal_controller.rb

```diff
@@ -0,0 +1,30 @@
+class InternalController < ApplicationController
+  before_action :authenticate
+  helper_method :publisher?
+
+protected
+
+  def publisher?
+    session[:publisher]
+  end
+
+private
+
+  def set_account_role(role)
+    session[role] = true
```

Comment:
I think having multiple session names for each role could lead to weirdness if you log in/out as a publisher/author as you could end up with `session[:author]` and `session[:publisher]` both `true` if you're not careful/clean up. I think having the session attribute static and the value variable would be less error prone:

```
session[:role] = :author
```